--- libzypp-17.31.9/cmake/modules/FindGpgme.cmake.orig	2023-03-31 18:39:00.000000000 +0200
+++ libzypp-17.31.9/cmake/modules/FindGpgme.cmake	2023-04-02 15:15:24.414928189 +0200
@@ -213,14 +213,14 @@ else() # not WIN32
     set( GPGME_PTHREAD_FOUND false )
     set( GPGME_PTH_FOUND     false )
 
-    find_program( _GPGMECONFIG_EXECUTABLE NAMES gpgme-config )
+    find_program( _GPGMECONFIG_EXECUTABLE NAMES pkg-config )
 
     # if gpgme-config has been found
     if ( _GPGMECONFIG_EXECUTABLE )
 
       message( STATUS "Found gpgme-config at ${_GPGMECONFIG_EXECUTABLE}" )
 
-      exec_program( ${_GPGMECONFIG_EXECUTABLE} ARGS --version OUTPUT_VARIABLE GPGME_VERSION )
+      exec_program( ${_GPGMECONFIG_EXECUTABLE} ARGS --modversion gpgme OUTPUT_VARIABLE GPGME_VERSION )
 
       set( _GPGME_MIN_VERSION "1.1.7" )
 
@@ -232,17 +232,17 @@ else() # not WIN32
 
         message( STATUS "Found gpgme v${GPGME_VERSION}, checking for flavours..." )
 
-        exec_program( ${_GPGMECONFIG_EXECUTABLE} ARGS                  --libs OUTPUT_VARIABLE _gpgme_config_vanilla_libs RETURN_VALUE _ret )
+        exec_program( ${_GPGMECONFIG_EXECUTABLE} ARGS                  --libs gpgme OUTPUT_VARIABLE _gpgme_config_vanilla_libs RETURN_VALUE _ret )
 	if ( _ret )
 	  set( _gpgme_config_vanilla_libs )
 	endif()
 
-        exec_program( ${_GPGMECONFIG_EXECUTABLE} ARGS --thread=pthread --libs OUTPUT_VARIABLE _gpgme_config_pthread_libs RETURN_VALUE _ret )
+        exec_program( ${_GPGMECONFIG_EXECUTABLE} ARGS --libs gpgme OUTPUT_VARIABLE _gpgme_config_pthread_libs RETURN_VALUE _ret )
 	if ( _ret )
 	  set( _gpgme_config_pthread_libs )
 	endif()
 
-        exec_program( ${_GPGMECONFIG_EXECUTABLE} ARGS --thread=pth     --libs OUTPUT_VARIABLE _gpgme_config_pth_libs     RETURN_VALUE _ret )
+        exec_program( ${_GPGMECONFIG_EXECUTABLE} ARGS --libs gpgme OUTPUT_VARIABLE _gpgme_config_pth_libs     RETURN_VALUE _ret )
 	if ( _ret )
 	  set( _gpgme_config_pth_libs )
 	endif()
@@ -256,7 +256,7 @@ else() # not WIN32
 
         if ( _gpgme_config_vanilla_libs OR _gpgme_config_pthread_libs OR _gpgme_config_pth_libs )
 
-          exec_program( ${_GPGMECONFIG_EXECUTABLE} ARGS --cflags OUTPUT_VARIABLE _GPGME_CFLAGS )
+          exec_program( ${_GPGMECONFIG_EXECUTABLE} ARGS --cflags gpgme OUTPUT_VARIABLE _GPGME_CFLAGS )
 
           if ( _GPGME_CFLAGS )
             string( REGEX REPLACE "(\r?\n)+$" " " _GPGME_CFLAGS  "${_GPGME_CFLAGS}" )
